File: p1/src/datablock.cpp
/* Michael LeMay, 1/13/19 */

#include "datablock.h"

/*Datablock implementation*/


Datablock::Datablock(int theSize) {
  size = theSize;
  /*Make sure you delete data before deleting datablock*/
  data = new double[size];
  blockTag = 0;
  blockValid = false;
}

Datablock::Datablock() {
  size = 0;
  data = new double[0];
  blockTag = 0;
  blockValid = false;
}

File: p1/src/datablock.py
# Michael LeMay, 1/19/19

import numpy as np

class Datablock:


    def __init__(self, size, id):
        """Initiates the block parameters. Base is a numpy array
        of a given size. I am storing doubles in this implementation.
        """
        self.size = size
        self.data = np.zeros(size)
        #self.valid = False
        self.id = id


    def printBlock(self):
        """Prints out all the doubles stored in this block"""
        print("Block is valid:", self.valid)
        for i in range(self.size):
            print("Datablock at", i, ": ", self.data[i])
